frame/name-service/src/lib.rs

```diff
@@ -0,0 +1,487 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A simple name service that can be used to give accounts friendly names.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use sp_runtime::traits::{Bounded, Convert, Saturating, Zero};
+	use sp_std::{convert::TryInto, vec::Vec};
+
+	use frame_support::traits::{
+		Currency, ExistenceRequirement, OnUnbalanced, ReservableCurrency, WithdrawReasons,
+	};
+
+	// The struct on which we build all of our Pallet logic.
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	type NameHash = [u8; 32];
+
+	type CommitmentHash = [u8; 32];
+
+	// Allows easy access our Pallet's `Balance` and `NegativeImbalance` type. Comes from `Currency`
+	// interface.
+	type BalanceOf<T> =
+		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+	type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
+		<T as frame_system::Config>::AccountId,
+	>>::NegativeImbalance;
+
+	// Your Pallet's configuration trait, representing custom external types and interfaces.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The Currency handler for the kitties pallet.
+		type Currency: ReservableCurrency<Self::AccountId>;
+
+		/// Convert the block number into a balance.
+		type BlockNumberToBalance: Convert<Self::BlockNumber, BalanceOf<Self>>;
+
+		/// The account where registration fees are paid to.
+		type RegistrationFeeHandler: OnUnbalanced<NegativeImbalanceOf<Self>>;
+
+		/// The deposit a user needs to make in order to commit to a name registration.
+		#[pallet::constant]
+		type CommitmentDeposit: Get<BalanceOf<Self>>;
+
+		/// The amount of blocks a user needs to wait after a Commitment before revealing.
+		#[pallet::constant]
+		type MinimumCommitmentPeriod: Get<Self::BlockNumber>;
+
+		/// The amount of blocks after a commitment is created for before it expires.
```

Comment:
```suggestion
		/// The amount of blocks after a commitment is created for before it expires.
		/// Must be larger than `MinimumCommitmentPeriod`.
```

Reply:
if there is a "must be larger", then we should add a compile time check.